scripts/libraries/mjpeg.py

```diff
@@ -0,0 +1,131 @@
+# This file is part of the OpenMV project.
+#
+# Copyright (c) 2023 Ibrahim Abdelkader <iabdalkader@openmv.io>
+# Copyright (c) 2023 Kwabena W. Agyeman <kwagyeman@openmv.io>
+#
+# This work is licensed under the MIT license, see the file LICENSE for details.
+#
+# This is an extension to the mjpeg C user-module.
+
+from umjpeg import *  # noqa
+import errno
+import re
+import socket
+
+
+class mjpeg_server:
+    def _valid_tcp_socket(self):  # private
+        if self._tcp_socket is None:
+            try:
+                s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                try:
+                    if hasattr(socket, "SO_REUSEADDR"):
+                        s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+                    s.bind(self._myaddr)
+                    s.listen(5)
+                    s.setblocking(True)
+                    self._tcp_socket, self._client_addr = s.accept()
+                except OSError:
+                    pass
+                s.close()
+            except OSError:
+                self._tcp_socket = None
+        return self._tcp_socket is not None
+
+    def _close_tcp_socket(self):  # private
+        if self._tcp_socket is not None:
+            self._tcp_socket.close()
+            self._tcp_socket = None
```

Comment:
This should be just socket not tcp_socket, even if it will somehow work over UDP.